app/players/player.py

```diff
@@ -0,0 +1,14 @@
+from abc import ABC, abstractmethod
+
+
+class Player(ABC):
+    def __init__(self, nickname: str) -> None:
+        self.nickname = nickname
+
+    @abstractmethod

```

Comment:
these methods will return specific types when implemented in child classes, so change annotation according to those